lib/extensions/ar_virtual.rb

```diff
@@ -405,6 +405,9 @@ def load_schema!
 
     def define_virtual_attribute(name, cast_type, uses: nil, arel: nil)
       attribute_types[name] = cast_type
+      if ActiveRecord::VERSION::MAJOR >= 5
+        _default_attributes[name] = ActiveRecord::Attribute.uninitialized(name, cast_type)
```

Comment:
This seems fine to me.. underscore notwithstanding, I think writing to `_default_attributes` is at a pretty similar level of internalness as `attribute_types`. (It bypasses the `define_default_attribute` private helper, but fundamentally, setting up those two hashes is [what `define_attribute` does](https://github.com/rails/rails/blob/4d9c591a91c353edc1aaed8fda6a5f702529b7e1/activerecord/lib/active_record/attributes.rb#L228-L229), and this method is intended to parallel that.)

No need for the version conditional, though -- unless it's fundamentally changed since I left it, this whole file is only expected to work on 5.0+.